File: Backend (Springboot)/Annapurna/src/main/java/com/annapurna/controller/AddressController.java
package com.annapurna.controller;

import com.annapurna.domain.Address;
import com.annapurna.domain.User;
import com.annapurna.dto.AddressDTO;
import com.annapurna.dto.AddressResponseDTO;
import com.annapurna.security.JwtProvider;
import com.annapurna.service.AddressService;
import com.annapurna.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/address")
@RequiredArgsConstructor
public class AddressController {

    private final AddressService addressService;
    private final UserService userService;
    private final JwtProvider jwtProvider;

    private User getCurrentUser(String authHeader) {
        String token = authHeader.substring(7); // Remove "Bearer "
        String email = jwtProvider.getEmailFromToken(token);
        return userService.getByEmail(email);
    }

    @PutMapping("/update")
    public ResponseEntity<String> updateAddress(@RequestHeader("Authorization") String authHeader,
                                                @RequestBody AddressDTO addressDTO) {
        User user = getCurrentUser(authHeader);
        addressService.updateAddress(user, addressDTO);
        return ResponseEntity.ok("Address updated successfully.");
    }
    
    
 

    @GetMapping("/my")
    public ResponseEntity<?> getMyAddress(@RequestHeader("Authorization") String authHeader) {
        User user = getCurrentUser(authHeader);
        Address address = addressService.getAddressByUser(user);

        if (address == null) {
            AddressResponseDTO empty = new AddressResponseDTO("", "", "", "", "");
            return ResponseEntity.ok(empty);
        }


        AddressResponseDTO dto = new AddressResponseDTO(
                address.getStreet(),
                address.getCity(),
                address.getState(),
                address.getPinCode(),
                address.getCountry()
        );

        return ResponseEntity.ok(dto);
    }




    
}

File: Backend (DNET)/Annapurna/Controllers/AddressController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Authorization;
using AnnapurnaAPI.Services;
using AnnapurnaAPI.Security;
using AnnapurnaAPI.DTOs;
using AnnapurnaAPI.Models;
using System.Security.Claims;

namespace AnnapurnaAPI.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    [Authorize]
    public class AddressController : ControllerBase
    {
        private readonly AddressService _addressService;
        private readonly UserService _userService;
        private readonly JwtService _jwtService;

        public AddressController(AddressService addressService, UserService userService, JwtService jwtService)
        {
            _addressService = addressService;
            _userService = userService;
            _jwtService = jwtService;
        }

        private async Task<User> GetCurrentUser()
        {
            var email = User.FindFirst(ClaimTypes.Email)?.Value;
            if (string.IsNullOrEmpty(email))
            {
                throw new InvalidOperationException("User not found");
            }
            return await _userService.GetByEmail(email) ?? throw new InvalidOperationException("User not found");
        }

        [HttpPut("update")]
        public async Task<IActionResult> UpdateAddress([FromBody] AddressDTO addressDTO)
        {
            try
            {
                var user = await GetCurrentUser();
                await _addressService.UpdateAddress(user, addressDTO);
                return Ok(new { message = "Address updated successfully." });
            }
            catch (Exception ex)
            {
                return BadRequest(new { message = ex.Message });
            }
        }

        [HttpGet("my")]
        public async Task<IActionResult> GetMyAddress()
        {
            try
            {
                var user = await GetCurrentUser();
                var address = await _addressService.GetAddressByUser(user);

                if (address == null)
                {
                    var empty = new AddressResponseDTO("", "", "", "", "");
                    return Ok(empty);
                }

                var dto = new AddressResponseDTO(
                    address.Street,
                    address.City,
                    address.State,
                    address.PinCode,
                    address.Country
                );

                return Ok(dto);
            }
            catch (Exception ex)
            {
                return BadRequest(new { message = ex.Message });
            }
        }
    }
}
